tests/textlab_api/textlab_api_test.py

```diff
@@ -13,3 +28,373 @@ def test_normalize_text(
+
+    :param load_test_data: The fixture providing the test data (see :meth:`~tests.conftest.load_test_data`)
+    :param login_role_user: The fixture providing the http client and the current role (see :meth:`~tests.conftest.login_role_user`)
+    :param settings: The fixture providing the django settings
+    :param httpserver: The fixture providing the dummy http server
+    """
+    # Setup a mocked Textlab API server with dummy responses
+    httpserver.expect_request("/user/login").respond_with_json({"token": "dummy"})
+    httpserver.expect_request("/benchmark/5").respond_with_json(
+        {"formulaHix": 0.0}, status=500
+    )
+
+    # Redirect call aimed at the Textlab API to the fake server
+    settings.TEXTLAB_API_URL = f"http://localhost:{httpserver.port}"
+
+    # Enable Textlab in the test page
+    Region.objects.filter(slug="augsburg").update(hix_enabled=True)
+
+    # Log the user in
+    client, _role = login_role_user
+
+    # Update page content
+    edit_page = reverse(
+        "edit_page",
+        kwargs={
+            "region_slug": "augsburg",
+            "language_slug": "de",
+            "page_id": 2,
+        },
+    )
+
+    content_before_change = Page.objects.get(id=2).get_translation("de").content
+    assert content_before_change != ""
+
+    edit_page_data = {
+        "title": "Neuer Titel",
+        "content": content_before_change,
+        "mirrored_page_region": "",
+        "_ref_node_id": 3,
+        "_position": "right",
+        "status": status.PUBLIC,
+    }
+
+    response = client.post(edit_page, edit_page_data)
+
+    assert response.status_code == 302
+    assert response.headers.get("Location") == edit_page
+
+    # Check the updated HIX score
+    page_translation = Page.objects.get(id=2).get_translation("de")
+
+    assert page_translation.hix_score == None
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("login_role_user", [EDITOR], indirect=True)
+def test_hix_response_400(
+    load_test_data: None,
+    login_role_user: tuple[Client, str],
+    settings: SettingsWrapper,
+    httpserver: HTTPServer,
+) -> None:
+    """
+    Check that the HIX score is requested but response 400 received. Hix score is not updated
+
+    :param load_test_data: The fixture providing the test data (see :meth:`~tests.conftest.load_test_data`)
+    :param login_role_user: The fixture providing the http client and the current role (see :meth:`~tests.conftest.login_role_user`)
+    :param settings: The fixture providing the django settings
+    :param httpserver: The fixture providing the dummy http server
+    """
+    # Setup a mocked Textlab API server with dummy responses
+    httpserver.expect_request("/user/login").respond_with_json({"token": "dummy"})
+    httpserver.expect_request("/benchmark/5").respond_with_json(
+        {"formulaHix": 15.48920568}, status=400
```

Comment:
Also it doesn't make much difference in the current case, but normally error responses (400, 500) imply that the request was not fulfilled.
So we are unlikely to get the error status code and the successful hix calculation result in the same response.
It would be more natural to return smthg like:
`{"message": "Error occurred"}`

Reply:
Thank you for the idea with respond_with_handler(). Just pushed the update.